### PR TITLE
Update JS APIs and clean up some markdown formatting

### PR DESCRIPTION
**Pull request by na--**
Closes https://github.com/grafana/k6-docs/issues/491 and closes https://github.com/grafana/k6-docs/issues/492

### CODE REVIEW

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md [hunk 1]
```diff
@@ -6,10 +6,12 @@ excerpt: 'Used for setting various WebSocket request-specific parameters such as
...
-| `Params.tags`         | object | Key-value pairs where the keys are names of tags and the values are tag values. Response time metrics generated as a result of the request will have these tags added to them, allowing the user to filter out those results specifically, when looking at results data.                                                                                                                                                                                                                                                                                                                          |
+| Name                  | Type   | Description |
+| --------------------- | ------ | ----------- |
+| `Params.compression`  | string | Empty by default, but when specified, WebSocket compression will be enabled with the specified algorithm. The only currently supported algorithm is `deflate`. |
```
Comment (inancgumus) [inline src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md:11]: Empty by default, thus disabled. WebSocket compression will only be enabled when an algorithm is specified. The only currently supported algorithm is `deflate`.

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md [hunk 2]
```diff
@@ -6,10 +6,12 @@ excerpt: 'Used for setting various WebSocket request-specific parameters such as
...
+| `Params.jar`          | [http.CookieJar](https://k6.io/docs/javascript-api/k6-http/cookiejar/) | The cookie jar that will be used when making the initial HTTP request to establish the WebSocket connection. If empty, the [default VU cookie jar](https://k6.io/docs/javascript-api/k6-http/cookiejar-method) will be used. |
+| `Params.headers`      | object | Object with key-value pairs representing custom headers the user would like to add to the request. |
+| `Params.tags`         | object | Key-value pairs where the keys are names of tags and the values are tag values. Response time metrics generated as a result of the request will have these tags added to them, allowing the user to filter out those results specifically, when looking at results data. |
```
Comment (inancgumus) [inline src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md:14]: Key-value pairs where the keys are names of tags and the values are tag values. Response time metrics generated from the request have these tags attached, allowing the user to filter the results data.

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md [hunk 3]
```diff
@@ -6,10 +6,12 @@ excerpt: 'Used for setting various WebSocket request-specific parameters such as
...
+| Name                  | Type   | Description |
+| --------------------- | ------ | ----------- |
+| `Params.compression`  | string | Empty by default, but when specified, WebSocket compression will be enabled with the specified algorithm. The only currently supported algorithm is `deflate`. |
+| `Params.jar`          | [http.CookieJar](https://k6.io/docs/javascript-api/k6-http/cookiejar/) | The cookie jar that will be used when making the initial HTTP request to establish the WebSocket connection. If empty, the [default VU cookie jar](https://k6.io/docs/javascript-api/k6-http/cookiejar-method) will be used. |
+| `Params.headers`      | object | Object with key-value pairs representing custom headers the user would like to add to the request. |
```
Comment (inancgumus) [inline src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md:13]: Custom headers in key-value pairs the user would like to add to the request. Keys are header names, and values are header values.

**Review by codebien — APPROVED** [review]

**Review by inancgumus — COMMENTED** [review]
LGTM.

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md (hunk 1 above)
Comment (na--) [reply src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md:11]: :thinking: WDYT about:
> WebSocket compression algorithm. It's empty by default, thus disabled, and will only be enabled when this option is specified. The only currently supported algorithm is `deflate`.

Comment (inancgumus) [reply src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md:11]: Better.

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md (hunk 2 above)
Comment (na--) [reply src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md:14]: > Key-value [custom metric tags](https://k6.io/docs/using-k6/tags-and-groups/#user-defined-tags), where the keys are names of tags and the values are tag values. [Metrics samples generated from the WebSocket connection](https://k6.io/docs/javascript-api/k6-ws/socket/#websocket-built-in-metrics) will have these tags attached, allowing users to filter the results data or set [thresholds on sub-metrics](https://k6.io/docs/using-k6/thresholds/#thresholds-on-tags).

:question:

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md (hunk 3 above)
Comment (na--) [reply src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md:13]: > Custom HTTP headers in key-value pairs the user would like to add to the initial HTTP request that establishes the WebSocket connection. Keys are header names, and values are header values.

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md (hunk 1 above)
Comment (na--) [reply src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md]: Resolved by https://github.com/grafana/k6-docs/pull/495/commits/268caeb8f30c7e52cd6b30649973a5d657bf8516

Comment (MStoykov) [reply src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md]: I do think that the options documentation should:
1. first and foremost explain what it is for
2. what are the possible values
3. what is the default

in that order.
So for me @na-- suggestion is better, but something like 
> compression algorithm to be used by the WebSocket connection. The only supported algorithm is `deflate` and if the option is left unset it defaults to no compression.  

is even more concise IMO.

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md (hunk 3 above)
Comment (inancgumus) [reply src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md]: Better ;)

**Review by inancgumus — APPROVED** [review]

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md (hunk 2 above)
Comment (inancgumus) [reply src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md]: [Custom metric tags](https://k6.io/docs/using-k6/tags-and-groups/#user-defined-tags) in key-value pairs where the keys are names of tags and the values are tag values. [The WebSocket connection will generate metrics samples](https://k6.io/docs/javascript-api/k6-ws/socket/#websocket-built-in-metrics) with these tags attached, allowing users to filter the results data or set [thresholds on sub-metrics](https://k6.io/docs/using-k6/thresholds/#thresholds-on-tags).

**Review by codebien — APPROVED** [review]

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md (hunk 2 above)
Comment (na--) [reply src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md]: Resolved by https://github.com/grafana/k6-docs/pull/495/commits/6f14bf3b21b18371e8a6a7aa8759b11a65e3c225

File: src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md (hunk 1 above)
Comment (na--) [reply src/data/markdown/docs/02 javascript api/11 k6-ws/20-Params.md]: Agree, resolved by https://github.com/grafana/k6-docs/pull/495/commits/a23595697396c19921c989b142830213c683a6d5, but please don't comment in resolved PR discussions (or unresolve them if you do), I almost missed this.

**Pull request merged**